src/runtime/pickle_runner.ts

```diff
@@ -395,9 +430,69 @@ export default class PickleRunner {
         },
       })
     }
-    return await this.invokeStep(
-      testStep.pickleStep,
-      testStep.stepDefinitions[0]
+
+    let cumulatedStepResult
```

Comment:
Okay think this needs to be re-structured to something like:
```
// At this point lets run all stepHooks + step
const beforeStepHookResults = await this.runStepHooks(/* */); // hooks do not receive "result" in parameter
const const stepResult = await this.invokeStep(/* */);
const afterStepHookResults = await this.runStepHooks(/* */); // hooks receive stepResult as "result" in parameter (not cumulatedResult)
// Then accumulate result, use query to find the worst result and sum the durations into it
```

Sorry for the rework here. Just trying to figure out what is both simple and easy to explain to users.